dio/lib/src/adapters/browser_adapter.dart

```diff
@@ -67,18 +67,34 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
     final completer = Completer<ResponseBody>();
 
     xhr.onLoad.first.then((_) {
-      final Uint8List body = (xhr.response as ByteBuffer).asUint8List();
-      completer.complete(
-        ResponseBody.fromBytes(
-          body,
-          xhr.status!,
-          headers: xhr.responseHeaders.map((k, v) => MapEntry(k, v.split(','))),
-          statusMessage: xhr.statusText,
-          isRedirect: xhr.status == 302 ||
-              xhr.status == 301 ||
-              options.uri.toString() != xhr.responseUrl,
-        ),
-      );
+      if (options.blobUrl) {
+        completer.complete(
+          ResponseBody.fromString(
+            Url.createObjectUrl(xhr.response),
```

Comment:
Any chance that this could fail? Maybe another `catch` here would be good.

Reply:
I tested and did not encounter exception !